src/server/pfs/cmds/cmds.go

```diff
@@ -669,21 +670,35 @@ Any pachctl command that can take a Commit ID, can take a branch name instead.`,
 			if len(provenance) != 0 && trigger.Branch != "" {
 				return errors.Errorf("cannot use provenance and triggers on the same branch")
 			}
-			if (trigger.CronSpec != "" || trigger.Size_ != "" || trigger.Commits != 0) &&
-				trigger.Branch == "" {
+			if (trigger.CronSpec != "" || trigger.Size_ != "" || trigger.Commits != 0) && trigger.Branch == "" {
 				return errors.Errorf("trigger condition specified without a branch to trigger on, specify a branch with --trigger")
 			}
+			if proto.Equal(trigger, &pfs.Trigger{}) {
+				trigger = nil
+			}
 			c, err := client.NewOnUserMachine("user")
 			if err != nil {
 				return err
 			}
 			defer c.Close()
 
+			var headCommit *pfs.Commit
+			if head != "" {
+				if uuid.IsUUIDWithoutDashes(head) {
+					headCommit = branch.Repo.NewCommit("", head)
+				} else {
+					headCommit, err = cmdutil.ParseCommit(head)
+					if err != nil {
+						return err
+					}
+				}
+			}
+
 			return txncmds.WithActiveTransaction(c, func(c *client.APIClient) error {
 				_, err := c.PfsAPIClient.CreateBranch(
 					c.Ctx(),
 					&pfs.CreateBranchRequest{
-						Head:       branch.Repo.NewCommit("", head),
```

Comment:
This was broken - it was specifying a head commit even if we weren't updating the branch head.